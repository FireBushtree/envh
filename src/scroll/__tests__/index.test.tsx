import * as React from 'react';
import { render } from '@testing-library/react';
import Scroll from '../scroll';

const loadingContainerClassName = '.eh-scroll-loading-container';
const requestDoneClassName = '.eh-scroll-loading__done-text';

describe('<Scroll />', () => {
  test('should render default', () => {
    const { container } = render(<Scroll />);
    expect(container).toMatchSnapshot();
  });

  test('hasRequestText prop', () => {
    const { container, rerender } = render(<Scroll />);

    let loadingContainer = container.querySelector(loadingContainerClassName);
    expect(loadingContainer).toBeTruthy();

    rerender(<Scroll hasRequestText={false} />);
    loadingContainer = container.querySelector(loadingContainerClassName);
    expect(loadingContainer).toBeFalsy();
  });

  test('requestDone prop', () => {
    const { container, rerender } = render(<Scroll />);
    let textContainer = container.querySelector(requestDoneClassName);
    expect(textContainer).toBeFalsy();

    rerender(<Scroll requestDone />);
    textContainer = container.querySelector(requestDoneClassName);
    expect(textContainer).toBeTruthy();
  });
});
