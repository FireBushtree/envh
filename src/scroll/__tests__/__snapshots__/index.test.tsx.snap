// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Scroll /> should render default 1`] = `
<div>
  <div
    class="eh-scroll"
  >
    <div>
      <div
        class="eh-scroll-loading-container"
      >
        <div
          class="eh-spin eh-spin-spinning eh-spin-inline"
        >
          <div
            class="eh-spin-roller"
          >
            <div />
          </div>
          <div
            class="eh-spin-mask"
          />
          <div
            class="eh-spin-children"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
